CoreLayersView/src/au/gov/asd/tac/constellation/views/layers/components/LayerTitlePane.java

```diff
@@ -0,0 +1,274 @@
+        enabled.setSelected(value);
+        enabled.selectedProperty().addListener(enabledChanged);
+        recolourLayer();
+    }
+
+    /**
+     * Set the current vertex query to the given query and check if it is valid
+     *
+     * @param query
+     */
+    public void setVxQuery(final String query) {
+        vxQuery.setValidity(true);
+        vxQuery.setQuery(query);
+        recheckValidity();
+        
+        if (!isValid) {
+            // when vx is invalid, 
+            vxQuery.setValidity(false);
+        }
+    }
+
+    /**
+     * Set the current transaction query to the given query and check if it is valid
+     *
+     * @param query
+     */
+    public void setTxQuery(final String query) {
+        txQuery.setValidity(true);
+        txQuery.setQuery(query);
+        recheckValidity();
+        if (!isValid) {
+            // when tx is invalid, 
+            txQuery.setValidity(false);
+        }
+    }
+
+    /**
+     * Set the description of the current query
+     *
+     * @param description
+     */
+    public void setDescription(final String description) {
+        descinput.setQuery(description);
+        final String displayedLayerName = StringUtils.truncate(description, MAX_DISPLAYED_CHARS);
+        label.setText(StringUtils.isBlank(description) ? String.format("%d", layerId) : String.format("%-2.2s - %s",String.valueOf(layerId), displayedLayerName));
+    }
+
+    /**
+     * Set the colour of the current layer to the selected style when selected or to the invalid style when a query is invalid
+     */
+    private void recolourLayer() {
+        if (enabled.isSelected()) {
+            getStyleClass().add(SELECTED_STYLE);
+        } else if (!isValid) {
+            // not selected and invalid
+            while (getStyleClass().contains(SELECTED_STYLE)) {
+                getStyleClass().remove(SELECTED_STYLE);
+            }
+            getStyleClass().add(INVALID_STYLE);
+            enabled.setDisable(true);
+            
+        } else {
+            // not selected, valid
+            while (getStyleClass().contains(SELECTED_STYLE) || getStyleClass().contains(INVALID_STYLE)) {
+                getStyleClass().remove(SELECTED_STYLE);
+                getStyleClass().remove(INVALID_STYLE);
+            }
+            enabled.setDisable(false);
+        }
+    }
+    
+    /**
+     * Delete this layer
+     */
+    private void delete() {
+        LayersViewController.getDefault().deleteLayer(layerId);
+    }
+    
+    public boolean isValid() {
+        return isValid;
+    }
+    
+    public void setValidity(final boolean isValid) {
+        this.isValid = isValid;
+        recolourLayer();
+    }
+    
+    public void recheckValidity() {
+        boolean validity = query.getQueryString() == null || ExpressionUtilities.testQueryValidity(query.getQueryString());
+        this.isValid = validity;
+        setValidity(validity);
```

Comment:
given `setValidity(validity)` will do exactly what the line above it is doing, I think that line above can be removed